pkg/apis/eventing/v1beta1/eventtype_conversion.go

```diff
@@ -61,7 +67,7 @@ func (sink *EventType) ConvertFrom(ctx context.Context, obj apis.Convertible) er
 			Source:      source.Spec.Source,
 			Schema:      source.Spec.Schema,
 			SchemaData:  source.Spec.SchemaData,
-			Broker:      source.Spec.Broker,
+			Broker:      source.Spec.Reference.Name,
```

Comment:
What if this has Kind InMemoryChannel?

Reply:
I wouldn't expect that have it set at all

Reply:
that's a separate step.
first is making it just KRef 

Channel is different issue in the project, but a separate aspect of the exercise

Reply:
Let me expand a bit, here we're converting a v1beta2 object to a v1beta1 object.

In v1beta2, every Kind (generally GVK) is accepted, so having broker set to the name of an object of different Kind has no meaning and is wrong because if I look it up and it doesn't exist

Reply:
So, we could just ingore `broker` field _if_ the `ref` is NOT a broker.

I can change that. 

Does that solve your point? Or do you mean different ?

Reply:
Doesn't that cause a lossy conversion?

Reply:
Ideally, we can somehow store reference in v1beta1 as well

Reply:
why not? and do you solve the lossy conversion we have now?